### PR TITLE
Add `ref` missing parameter in `content` endpoint

### PR DESCRIPTION
**Pull request by Palleas**
The current API doesn't allow you to get a content at a specific ref and this fixes it.

### CODE REVIEW

File: Tentacle/Client.swift [hunk 1]
```diff
@@ -213,13 +213,16 @@ public final class Client {
...
-            case let .content(owner, repository, path):
+            case let .content(owner, repository, path, _):
                 return "/repos/\(owner)/\(repository)/contents/\(path)"
             }
         }
         
         internal var queryItems: [URLQueryItem] {
-            return []
+            switch self {
+            case let .content(_, _, _, ref): return [URLQueryItem(name: "ref", value: ref)]
```
Comment (mdiep) [inline Tentacle/Client.swift]: Does this behave correctly if `ref` is `nil`?

**Review by mdiep — COMMENTED** [review]
Is there a way that we can test this? Maybe testing that the `Endpoint` has the correct parameters? Or maybe that's only worth doing if we need to special case a `nil` ref.

**Comment by Palleas** [comment]
Rebased and Added some tests for the Endpoint enum. ✌🏼

File: Sources/Tentacle/Client.swift [hunk 2]
```diff
@@ -337,12 +343,13 @@ public final class Client {
...
-    public func content(atPath path: String, in repository: Repository) -> SignalProducer<(Response, Content), Error> {
-        return fetchOne(.content(owner: repository.owner, repository: repository.name, path: path))
+    public func content(atPath path: String, in repository: Repository, at ref: String? = nil) -> SignalProducer<(Response, Content), Error> {
+        return fetchOne(.content(owner: repository.owner, repository: repository.name, path: path, ref: ref))
     }
 
-    public func create(file: File, atPath path: String, in repository: Repository) -> SignalProducer<(Response, FileResponse), Error> {
-        return send(file, to: .content(owner: repository.owner, repository: repository.name, path: path), using: .put)
+    /// Create a file in a repository
+    public func create(file: File, atPath path: String, in repository: Repository, in branch: String? = nil) -> SignalProducer<(Response, FileResponse), Error> {
```
Comment (ikesyo) [inline Sources/Tentacle/Client.swift]: These new parameters above should follow the API Design Guidelines: https://swift.org/documentation/api-design-guidelines/

> Compensate for weak type information to clarify a parameter’s role.

**Comment by Palleas** [comment]
@ikesyo is it better now? I used `inBranch` and `atRef` 🤔

**Review by mdiep — APPROVED** [review]

**Pull request merged**